internal/docstore/firedocstore/fs.go

```diff
@@ -357,6 +373,19 @@ func (c *collection) runWrites(ctx context.Context, actions []*driver.Action) er
 		}
 		newNames[i] = nn
 		pws = append(pws, ws...)
+		if opts.BeforeDo != nil {
+			asFunc := func(i interface{}) bool {
+				p, ok := i.(*[]*pb.Write)
```

Comment:
I think this should be the CommitRequest instead.

Reply:
That's right, I somehow missed that type.

PTAL